Actually the updated code does not work when building with soft PWM (currently only MicroBit and Arduino Primo) because of multiple definition of `TIMER1_IRQHandler' (the other from drivers/ticker.c).

--- a/ports/nrf/modules/machine/timer.c
+++ b/ports/nrf/modules/machine/timer.c
@@ -53,7 +53,9 @@ STATIC mp_obj_t machine_timer_callbacks[] = {
 
 STATIC const machine_timer_obj_t machine_timer_obj[] = {
     {{&machine_timer_type}, NRFX_TIMER_INSTANCE(0)},
-#if !defined(MICROPY_PY_MACHINE_SOFT_PWM) || (MICROPY_PY_MACHINE_SOFT_PWM == 0)
+#if MICROPY_PY_MACHINE_SOFT_PWM
+    { },
+#else
     {{&machine_timer_type}, NRFX_TIMER_INSTANCE(1)},
 #endif
     {{&machine_timer_type}, NRFX_TIMER_INSTANCE(2)},

